jsonschema-core.xml

```diff
@@ -313,43 +313,70 @@
+                        </dd>
+                        <dt>Assertion keywords</dt>
                         <dd>
-                            control schema identification through setting a IRI
-                            for the schema and/or changing how the base IRI is determined
+                            Keywords that may decide an instance is invalid.
                         </dd>
-                        <dt>assertions</dt>
+                        <dt>Annotation keywords</dt>
                         <dd>
-                            produce a boolean result when applied to an instance
+                            Keywords that produce output generated from an instance.
                         </dd>
-                        <dt>annotations</dt>
+                        <dt>Applicator keywords</dt>
                         <dd>
-                            attach information to an instance for application use
+                            A keyword that has one or more subschemas,
+                            that are applied against the instance,
+                            or some value within the instance,
+                            and logically aggregated together.
                         </dd>
-                        <dt>applicators</dt>
+                        <dt>Reserved keywords</dt>
                         <dd>
-                            apply one or more subschemas to a particular location
-                            in the instance, and combine or modify their results
+                            Keywords that are permitted in a schema,
+                            but that do not do anything by themselves,
+                            though their value may be read by or referred to by another keyword.
                         </dd>
-                        <dt>reserved locations</dt>
+                        <dt>Arguments</dt>
                         <dd>
-                            do not directly affect results, but reserve a place
-                            for a specific purpose to ensure interoperability
+                            Arguments are properties in a schema that are used by an adjacent keyword,
+                            and whose meaning may only be defined in the presence of that keyword.
                         </dd>
                     </dl>
                     <t>
-                        Keywords may fall into multiple categories, although applicators
-                        SHOULD only produce assertion results based on their subschemas'
-                        results.  They should not define additional constraints independent
-                        of their subschemas.
+                        Some properties may be classified into multiple categories.
+                        Some keywords are also arguments to other keywords,
+                        and applicator keywords are both assertions and annotations.
```

Comment:
Applicator keywords _may_ be assertive and annotative, but not necessarily so.  `anyOf` isn't annotative.  Its subschemas produce annotations, but the keyword itself does not.